hooks/src/index.js

```diff
@@ -193,7 +193,7 @@ export function useReducer(reducer, initialState, init) {
 				// When we have no updated hooks in the component we invoke the previous SCU or
 				// traverse the VDOM tree further.
 				if (allHooksEmpty) {
-					return prevScu ? prevScu(p, s, c) : true;
+					return prevScu ? prevScu.call(this, p, s, c) : true;
```

Comment:
Which `this` are you trying to propagate? This one is saved when the `useReducer()` call happened, not when `component.shouldComponentUpdate()` was called.

Reply:
whoooops, good catch.